tests/Geocoder/Tests/Provider/GoogleMapsTest.php

```diff
@@ -427,4 +427,64 @@ public function testGeocodeWithInvalidClientIdAndKeyNoSsl()
         $provider = GoogleMaps::business($this->getAdapter(), 'foo', 'bogus', null, null, false);
         $provider->geocode('Columbia University');
     }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidCountry()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'ru', 'postal_code' => '75020'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidZipCode()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'fr', 'postal_code' => '00000'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    public function testGeocodeWithValidComponentFilters()
```

Comment:
@mvcaaa i did something silimar for googles `result_type` & `location_type` and wrote a unittest that checks the url.
Check out here: https://github.com/arubacao/Geocoder/blob/8f322d551b486c8c3bf6fd7fe84caaec6364797e/tests/Geocoder/Tests/Provider/GoogleMapsTest.php#L304